x-pack/plugins/observability_solution/observability_ai_assistant/server/service/inference_endpoint.ts

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { errors } from '@elastic/elasticsearch';
+import { ElasticsearchClient } from '@kbn/core-elasticsearch-server';
+import { Logger } from '@kbn/logging';
+import moment from 'moment';
+
+export const AI_ASSISTANT_KB_INFERENCE_ID = 'ai_assistant_kb_inference';
+
+export interface CreateInferenceEndpointResponse {
+  inference_id: string;
+  task_type: string;
+  service: string;
+  service_settings: { num_allocations: number; num_threads: number; model_id: string };
+  chunking_settings: { strategy: string; max_chunk_size: number; sentence_overlap: number };
+}
+
+export async function createInferenceEndpoint({
+  esClient,
+  logger,
+  modelId = '.elser_model_2',
+}: {
+  esClient: {
+    asCurrentUser: ElasticsearchClient;
+  };
+  logger: Logger;
+  modelId: string | undefined;
+}) {
+  try {
+    logger.debug(`Creating inference endpoint "${AI_ASSISTANT_KB_INFERENCE_ID}"`);
+    return await esClient.asCurrentUser.transport.request<CreateInferenceEndpointResponse>(
+      {
+        method: 'PUT',
+        path: `_inference/sparse_embedding/${AI_ASSISTANT_KB_INFERENCE_ID}`,
+        body: {
+          service: 'elasticsearch',
+          service_settings: {
+            model_id: modelId,
+            num_allocations: 1,
```

Comment:
You'll want to set up [adaptive allocations](https://www.elastic.co/guide/en/elasticsearch/reference/8.16/infer-service-elser.html) here so that this:
A) auto-scales
B) Can scale to zero in Serverless (so we can enable this in ES3 :))

Reply:
Also consider using the APIs exported by the ML plugin instead of directly calling Elasticsearch endpoints. That'll help with adjusting to changes to this API.

Reply:
Good points! I've replaced the static allocations with adaptive allocations.

I looked at the ML code but couldn't find any appropriate APIs. And their implementation honestly looked overly complex anyway

Reply:
@dgieselaar I've changed the `ready` flag to include the allocation count:

```ts
const ready = deploymentState === 'started' && allocationState === 'fully_allocated' && allocationCount > 0;
```